backend/main/server/helpers.go

```diff
@@ -592,6 +596,127 @@ func (h *Helpers) createProposal(p models.Proposal) (models.Proposal, errorRespo
 	return p, nilErr
 }
 
+func (h *Helpers) createDraftProposal(c models.Community, p models.Proposal) (models.Proposal, errorResponse) {
+	canUserCreateProposal := c.CanUserCreateProposal(
+		h.A.DB,
+		h.A.FlowAdapter,
+		p.Creator_addr,
+	)
+
+	if err := handlePermissionErrorr(canUserCreateProposal); err != nilErr {
+		return models.Proposal{}, err
+	}
+
+	defaultStrategy := "token-weighted-default"
+	draftStatus := "draft"
+	p.Strategy = &defaultStrategy
+	p.Status = &draftStatus
+
+	p = h.setNullFieldsToEmpty(p)
+
+	if err := p.CreateProposal(h.A.DB); err != nil {
+		return models.Proposal{}, errIncompleteRequest
+	}
+
+	return p, nilErr
+}
+
+func (h *Helpers) setNullFieldsToEmpty(p models.Proposal) models.Proposal {
+	emptyString := ""
+	emptyTime := time.Time{}
+	emptyFloat := float64(0)
+	emptyUint64 := uint64(0)
+
+	if &p.Start_time == nil {
+		p.Start_time = emptyTime
+	}
+	if &p.End_time == nil {
+		p.End_time = emptyTime
+	}
+	if p.Min_balance == nil {
+		p.Min_balance = &emptyFloat
+	}
+	if p.Max_weight == nil {
+		p.Max_weight = &emptyFloat
+	}
+	if p.Strategy == nil {
+		p.Strategy = &emptyString
+	}
+	if p.Cid == nil {
+		p.Cid = &emptyString
+	}
+	if p.Block_height == nil {
+		p.Block_height = &emptyUint64
+	}
+	if p.Status == nil {
+		p.Status = &emptyString
+	}
+	if p.Created_at == nil {
+		p.Created_at = &emptyTime
+	}
+
+	return p
+}
+
+func (h *Helpers) updateDraftProposal(p models.Proposal) (models.Proposal, errorResponse) {
+	if p.Voucher != nil {
+		if err := h.validateUserViaVoucher(p.Creator_addr, p.Voucher); err != nil {
+			return models.Proposal{}, errForbidden
+		}
+	} else {
+		if err := h.validateUser(p.Creator_addr, p.Timestamp, p.Composite_signatures); err != nil {
+			return models.Proposal{}, errForbidden
+		}
+	}
+
+	community, err := h.fetchCommunity(p.Community_id)
+	if err != nil {
+		return models.Proposal{}, errIncompleteRequest
+	}
+
+	canUserCreateProposal := community.CanUserCreateProposal(
+		h.A.DB,
+		h.A.FlowAdapter,
+		p.Creator_addr,
+	)
+
+	if err := handlePermissionErrorr(canUserCreateProposal); err != nilErr {
+		return models.Proposal{}, err
+	}
+
+	p = h.setNullFieldsToEmpty(p)
+
+	if err := p.UpdateDraftProposal(h.A.DB); err != nil {
+		return models.Proposal{}, errIncompleteRequest
+	}
+
+	return p, nilErr
+}
+
+func handlePermissionErrorr(result models.CanUserCreateProposalResponse) errorResponse {
+	// If user doesn't have permission, populate errorResponse
+	// with reason and error.
+	if !result.HasPermission {
+		errPermissionsResponse := errCreateProposalPermissions
+		errPermissionsResponse.Message = result.Reason
+		errPermissionsResponse.Details = result.Error.Error()
+		return errPermissionsResponse
+	}
+	return nilErr
+}
+
+func (h *Helpers) validateUpdateProposalStatus(status string) error {
+	validStatuses := []string{"draft", "pending", "published", "cancelled"}
```

Comment:
we can remove `pending` here, i dont think its ever used as a `status`, just a `computedStatus`